src/qibo/backends/pytorch.py

```diff
@@ -0,0 +1,574 @@
+import collections
+from typing import Union
+
+import numpy as np
+import torch
+
+from qibo import __version__
+from qibo.backends import einsum_utils
+from qibo.backends.npmatrices import NumpyMatrices
+from qibo.backends.numpy import NumpyBackend
+from qibo.config import raise_error
+from qibo.result import CircuitResult, MeasurementOutcomes, QuantumState
+
+torch_dtype_dict = {
+    "int": torch.int32,
+    "float": torch.float32,
+    "complex": torch.complex64,
+    "int32": torch.int32,
+    "int64": torch.int64,
+    "float32": torch.float32,
+    "float64": torch.float64,
+    "complex64": torch.complex64,
+    "complex128": torch.complex128,
+}
+
+
+class TorchMatrices(NumpyMatrices):
+    # Redefine parametrized gate matrices for backpropagation to work
+
+    def __init__(self, dtype):
+        super().__init__(dtype)
+        self.torch = torch
```

Comment:
I think doing `self.np = torch` will lead to the lesser possible amount of code rewriting